runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/FileWatcherTask.java

```diff
@@ -0,0 +1,147 @@
+package io.aklivity.zilla.runtime.engine.internal.registry;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_CREATE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_DELETE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
+import static org.agrona.LangUtil.rethrowUnchecked;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+
+public class FileWatcherTask extends WatcherTask
+{
+    private final Map<Path, byte[]> configHashes;
+    private WatchService watchService;
+
+    public FileWatcherTask(BiConsumer<URL, String> configChangeListener)
+    {
+        super(configChangeListener);
+        this.configHashes = new HashMap<>();
+        try
+        {
+            watchService = FileSystems.getDefault().newWatchService();
+        }
+        catch (IOException ex)
+        {
+            rethrowUnchecked(ex);
+        }
+
+    }
+
+    @Override
+    public boolean run()
+    {
+        try
+        {
+            while (true)
+            {
+                try
+                {
+                    final WatchKey key = watchService.take();
+
+                    for (WatchEvent<?> event : key.pollEvents())
+                    {
+                        final Path changed = ((Path) key.watchable()).resolve((Path) event.context());
+                        if (configHashes.containsKey(changed))
+                        {
+                            String configText = readConfigText(configURLs.get(changed));
+                            byte[] newConfigHash = computeHash(configText);
+                            if (!Arrays.equals(configHashes.get(changed), newConfigHash))
+                            {
+                                configHashes.put(changed, newConfigHash);
+                                configChangeListener.accept(configURLs.get(changed), configText);
+                            }
+                        }
+                    }
+                    key.reset();
+                }
+                catch (InterruptedException ex)
+                {
+                    watchService.close();
+                    break;
+                }
+            }
+        }
+        catch (IOException ex)
+        {
+            rethrowUnchecked(ex);
+        }
+        return true;
+    }
+
+    @Override
+    protected void doInitialConfiguration(URL configURL)
+    {
+        Path configPath = Paths.get(new File(configURL.getPath()).getAbsolutePath());
```

Comment:
Simplifies to `Path configPath = Paths.get(configURL.toURI()).getAbsolutePath();` ?

Reply:
Marked resolved but not applied?

Reply:
Ohh, yes. I don't know why I marked this as resolved. When we use config files in a jar, we get: 'java.nio.file.FileSystemNotFoundException' exception if trying to get the Path from the URI